### DISCUSSION TITLE
Is Chess Infinitum custom variant possible?

### DISCUSSION THREAD

**Discussion by joshxyzhimself** [fairy-stockfish/Fairy-Stockfish#633] [Q&A]
We are looking to try Fairy Stockfish for generating computer moves for the Chess Infinitum variant.

Live example can be played here: https://chess.mariuslab.com/infinitum/

A few highlights on the variation in the rules:

- Most rules are inherited from standard chess.
- It's a 10x10 board
  - **This seems possible with `maxFile`, `maxRank`, and `startFen`.**
- It has 8 squares gated / unusuable.
  - **This seems possible with `staticGatingRegion`, I need to compile the latest `master` branch.**
- There is a `Wizard` piece that moves like the `Queen` piece.
  - **This seems possible with the predefined piece types.**
- The `King` cannot capture the `Wizard` piece.
  - **Unsure how to do this yet, is this possible?**
- There are four (4) magic squares for each side that grants powers.
  - The regions: `e2 e3 f2 f3` for white, `e8 e9 f8 f9` for black.
  - The powers: If a player's piece is in one of the magic squares on his side, his piece can move like it exists on all unoccupied magic squares of his side.
  - The exceptions:The player's pieces do not gain the same powers when entering his opponent's magic squares.
  - **Unsure how to do this yet, is this possible?**

**Comment by ianfab** [comment]
> It's a 10x10 board
This seems possible with maxFile, maxRank, and startFen.

yes

> It has 8 squares gated / unusuable.
This seems possible with staticGatingRegion, I need to compile the latest master branch.

You just need to put `*` in the startFen in the places of the unavailable squares, no extra option is needed. See https://github.com/fairy-stockfish/Fairy-Stockfish/blob/bf310e8cbcd8fd171ef0a1b8a5144ea3681b0017/src/variants.ini#L839-L848 as an example.

> There is a Wizard piece that moves like the Queen piece.
This seems possible with the predefined piece types.

yes

> The King cannot capture the Wizard piece.
Unsure how to do this yet, is this possible?

If the king can capture other pieces but not this one specific piece, that is not supported

> There are four (4) magic squares for each side that grants powers

As a generalization of pawn double steps there is the possibility to give pieces extra movements on a set of squares, but it can only be defined as movements from the given square and not from other squares, so it does not cover your scenario.
